### DISCUSSION TITLE
Avoid cloning repositories if git repo is already present in basedir

### DISCUSSION THREAD

**Discussion by dmpe** [renovatebot/renovate#10023] [Request Help]
Hi there, 
we are using gitlab (runner) for our programming and have a git repo, which has several jobs. So, whenever a job is triggered, the runner starts, downloads the repo and executes steps in the job. 

Am I correct in my understanding, that renovate does its own git clone?

For example, when I pass 

```
renovate $CI_PROJECT_PATH
```

renovate will go and fetch that repository (based on my platform & token), unpacking it into $basedir. 

What I am looking for is to say e.g. 

```
renovate ./my-custom-helm-chart
```
whereby renovate avoids git clone, and "renovates" ` ./my-custom-helm-chart` folder, which might have some files "pre-processed" by the its own job.

Example:

```
sed -i ..... ./my-custom-helm-chartvalues.yaml
renovate ./my-custom-helm-chart
```

Thank you for any help.

**Comment by rarkins** [comment]
It's not supported, and nothjng like that planned. It would have a lot of complexity and edge cases. Please elaborate more on your use case so that we can understand it better and perhaps propose an alternative solution.

**Reply by dmpe** [reply]
Ok, interesting. 
Sure, in my particular case I have following `values.yaml.gotmpl` file, which I want to renovate using gitlab as a platfrorm, `helm-values` as manager and my private docker registry (registry.firm.com:9999) as a data source. 
The yaml file looks like this:

```
image:
  repository: {{ .Values.private_reg }}/sonatype/nexus3
  tag: v3.30.0
```

In order to deploy such yaml file to k8s cluster, we use internally helmfile and large shell script, which helps us to template such yaml. At runtime, `{{ .Values.private_reg }}` is replaced with our private registry, becoming `registry.firm.com:9999/sonatype/nexus3`. And here lies the problem - what I need is to preprocess such files e.g. with `sed` and only then run renovate. At the moment, when renovate runs, it clones the repository, but "only" sees `{{ .Values.private_reg }}`....

What would help me is to have (either):
- option to specify args (e.g. similarly to `jq`, which would be passed to the code below) - probably not feasable OR
- some sort of [preUpgradeTask](https://github.com/renovatebot/renovate/issues/8804), which would run once before renovate does anything with deps. It would be magical, if i could pass some shell script :)
- ? (not sure if there is anything else)

My current workaround would be probably to create a new branch (synched with master), which has `{{ .Values.private_reg }}` already replaced with `registry.firm.com:9999`... that branch would be then scaned. 
Let me know if there is some better way. 

Thanks.

**Reply by viceice** [reply]
You should use a regex manager to directly replace the version in your template.

if your repos are huge, you can play with [persistRepoData](https://docs.renovatebot.com/self-hosted-configuration/#persistrepodata), so renovate will not delete the local git copy and only fetches required commits.

**Reply by rarkins** [reply]
We are also considering pre-extract transforms: #7333

**Reply by dmpe** [reply]
Having this:
```
image:
  repository: {{ .Values.nexus }}/gitlab-ce/gitlab
  tag: v12.10-ee
```
you regex may look like 
```
(?<depName>(?<=repository: \{\{ \.Values\.nexus \}\}\/).+(?=.*))
```
However, given that RE2 does not support lookahead, back to square one
